### ISSUE TITLE
Nextjs - Source mapping failed: A source map could not be applied

### ISSUE THREAD

**Issue by charlimxm** [bugsnag/bugsnag-js#474]
Anyone using Next.js and has successfully managed to upload source maps to bugsnag?

I have successfully set up my next.config.js file to upload my source maps to bugsnag using webpack-bugsnag-plugins. 

However, bugsnag cannot seem to apply the source map to my code as seen in the screenshot below. Also, release stage is recorded as Production even though the issue is triggered from Staging (I've tried setting bugsnag releaseStage to process.env.NODE_ENV and it does not work).

![screen shot 2019-01-30 at 2 26 22 pm](https://user-images.githubusercontent.com/31029245/51962833-718bdb80-249c-11e9-899d-ecb96f12a2a2.png)

next.config.js
```
const fs = require('fs')
const withSass = require('@zeit/next-sass')
const { BugsnagSourceMapUploaderPlugin } = require('webpack-bugsnag-plugins')

module.exports = withSass({
  webpack: (config, { buildId, dev, isServer }) => {
    const oldEntry = config.entry
    config.entry = function () {
      return oldEntry().then(function (entries) {
        if (entries['main.js']) {
          entries['main.js'].unshift('babel-polyfill')
        }

        return entries
      })
    }

    config.module.rules = config.module.rules.map(rule => {
      if (rule.loader === 'babel-loader') {
        rule.options.cacheDirectory = false
      }
      return rule
    })

    config.devtool = 'source-map'
    for (const plugin of config.plugins) {
      if (plugin['constructor']['name'] === 'UglifyJsPlugin') {
        plugin.options.sourceMap = true
        break
      }
    }

    config.plugins.push(
      new BugsnagSourceMapUploaderPlugin({
        apiKey: 'af3a166c839bf6a0e9304c1bae9e5e24',
        publicPath: '*/',
        overwrite: true
      })
    )

    return config
  },
  generateBuildId: () => {
    const buildId = fs.readFileSync(process.cwd() + '/build/BUILD_ID', 'utf8')
    return buildId.trim()
  },
  poweredByHeader: false
})
```

**Comment by snmaynard** [comment]
Closing this issue in favor of a support ticket to support@bugsnag.com

**Comment by dallen4** [comment]
Was there any definitive solution for implementing source maps for Next.js. We use Next.js apps deployed on Zeit's Now platform and I can't for the life of me figure out the right way to do this. I implemented the `BugsnagSourceMapUploaderPlugin` from `webpack-bugsnag-plugins`, but the closest I get is the same error that @charlimxm experience. It seems to be trying to parse the files, but I'm guessing it might not be able to resolve them as the stracktrace starts referencing chunks which follow a different url path pattern than the page urls which include a unique build ID. I think this may be the "something specific to Next.js" that @mattdyoung was referring to. Anyway, any support would be appreciated so we can expedite our debugging of production errors :)

**Comment by mattdyoung** [comment]
For the benefit of anyone else following this, @dallen4 kindly put together a reproduction example for Bugsnag Support.

It appears there's an issue with the source maps generated by next-source-maps `0.0.3` where they don't match the deployed code. Updating to use the `0.0.4-canary.1` release allowed the generated source maps to be correctly applied.

**Comment by dallen4** [comment]
@mattdyoung and anyone who may need it, the example repo for configuring source map uploads to Bugsnag during the build and deployment process can be found [here](https://github.com/dallen4/bugsnag-sourcemaps-example). Please feel free to reach out for any clarifications.

**Comment by carloscuesta** [comment]
That's interesting @mattdyoung! Should we open an issue to the [`next-sourcemaps`](https://github.com/zeit/next-plugins/tree/master/packages/next-source-maps) repository ?

**Comment by mattdyoung** [comment]
There may already be an issue for it as it appears fixed in `0.0.4-canary.1`. What really needs to happen is for `next-source-maps` to release a version `0.0.4`.

**Comment by carloscuesta** [comment]
But we can try to kindly remind this, since the fix was pushed "15 months ago". So maybe reopening an issue to remind that this is a pain for some users will help on releasing this to the next version
